sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/FieldValueTypeInformation.java

```diff
@@ -118,6 +123,29 @@ public static FieldValueTypeInformation forField(Field field) {
         .build();
   }
 
+  public static <T extends AnnotatedElement & Member> String getNameOverride(
+      String original, T member) {
+    SchemaFieldName fieldName = member.getAnnotation(SchemaFieldName.class);
+    SchemaCaseFormat caseFormatAnnotation = member.getAnnotation(SchemaCaseFormat.class);
+    SchemaCaseFormat classCaseFormatAnnotation =
+        member.getDeclaringClass().getAnnotation(SchemaCaseFormat.class);
```

Comment:
I hate inheritance but I have to ask: what happens with

```
@SchemaCaseFormat(A)
class Parent {
  ... fields ...
}

@SchemaCaseFormat(B)
class Child extends Parent { }
```

I'm honestly not sure what should happen. I think in your case anything not overridden will get case format A. That seems right but just want to highlight it.

Reply:
Hm I didn't consider inheritance at all. I think your assessment about how this will behave is correct. I think this behavior would be fine, except in a context like this:

```java
class SomethingOutOfMyControl {
   ... fields ...
}

@SchemaCaseFormat(B)
class MyBeamObject extends SomethingOutOfMyControl { }
```

Not sure how often something like that comes up, but maybe that's enough reason to scope the case format to the class we're inferring the schema from.

Reply:
Worst cast, the person authoring `MyBeamObject` has to write a bunch of trivial overrides. I'm OK with that.